internal/operands/common-templates/reconcile_test.go

```diff
@@ -100,4 +102,61 @@ var _ = Describe("Common-Templates operand", func() {
 		Expect(err).ToNot(HaveOccurred())
 		ExpectResourceExists(newEditRole(), request)
 	})
+
+	It("should remove labels only from old templates", func() {
+		oldVersion := "v0.0.1"
+		err := request.Client.Create(request.Context, getTestTemplate(oldVersion, "1", namespace))
+		Expect(err).ToNot(HaveOccurred())
+		err = request.Client.Create(request.Context, getTestTemplate(oldVersion, "2", "test-namespace"))
+		Expect(err).ToNot(HaveOccurred())
+		err = request.Client.Create(request.Context, getTestTemplate(Version, "3", namespace))
+		Expect(err).ToNot(HaveOccurred())
+
+		_, err = operand.Reconcile(&request)
+		Expect(err).ToNot(HaveOccurred())
+
+		var oldTemplates templatev1.TemplateList
+		Expect(request.Client.List(request.Context, &oldTemplates)).ToNot(HaveOccurred())
```

Comment:
I usually write this with `err := ...` because the wording sounds off (Expect List ToNot HaveOccurred) but that's just my obsession with wording.